### PR TITLE
OpenShift service account support: NLS messages and unit tests

### PR DESCRIPTION
**Pull request by ayoho**

### CODE REVIEW

**Comment by ayoho** [comment]
#build

File: dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops [hunk 1]
```diff
@@ -94,6 +94,31 @@ KUBERNETES_USER_API_RESPONSE_DEFAULT_USER_ATTR_NOT_FOUND=CWWKS5381W: The social
...
 
+# 0=Exception message
+ERROR_INTROSPECTING_SERVICE_ACCOUNT=CWWKS5382E: The response from the service account user API cannot be processed: {0}
+ERROR_INTROSPECTING_SERVICE_ACCOUNT.explanation=Information about the service account cannot be obtained because another error occurred.
+ERROR_INTROSPECTING_SERVICE_ACCOUNT.useraction=If the message contains another error message, see the user action for that message. Otherwise, check the server logs for more error messages that might indicate where the other error occurred.
+
+# 0=HTTP response code, 1=Response body
+SERVICE_ACCOUNT_USER_API_BAD_STATUS=CWWKS5383E: The service account user API returned an unexpected [{0}] response code. Verify that the request to the API contains all of the required information and that the service account token is valid. The response from the API is [{1}].
+SERVICE_ACCOUNT_USER_API_BAD_STATUS.explanation=The service account user API did not return the expected status code.
```
Comment (helyarp) [inline dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops]: Please change "all of the required information" to "all required information" or, optionally, to "all the required information". This comment applies to the explanation and useraction.

File: dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops [hunk 2]
```diff
@@ -94,6 +94,31 @@ KUBERNETES_USER_API_RESPONSE_DEFAULT_USER_ATTR_NOT_FOUND=CWWKS5381W: The social
...
+SERVICE_ACCOUNT_USER_API_BAD_STATUS.useraction=Check the API response that is in the error message for more information. Verify that the request to the API contains all of the required information. Ensure that the service account token is valid.
+
+# 0=Response string, 1=Exception message
+RESPONSE_NOT_JSON=CWWKS5384E: The content of the response is not a valid JSON object. The full response is {0}. {1}
+RESPONSE_NOT_JSON.explanation=The response is expected to be a valid JSON object.
```
Comment (helyarp) [inline dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops]: Please change "is expected to be" to "must be".

File: dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops [hunk 3]
```diff
@@ -94,6 +94,31 @@ KUBERNETES_USER_API_RESPONSE_DEFAULT_USER_ATTR_NOT_FOUND=CWWKS5381W: The social
...
+RESPONSE_NOT_JSON.useraction=Verify that request was sent to the expected target. Ensure that the intended target of the request is capable of returning JSON data. Check the response to see whether more information is included.
+
+# 0=JSON key string, 1=Response string
+JSON_MISSING_KEY=CWWKS5385E: The JSON object that is provided is missing an expected key: [{0}]. The full JSON object is [{1}].
+JSON_MISSING_KEY.explanation=The key that is specified in the message is expected to be within the JSON data. The key might be missing, or it might be in an unexpected location.
```
Comment (helyarp) [inline dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops]: Please change "is expected to be" to "must be".

File: dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops [hunk 4]
```diff
@@ -94,6 +94,31 @@ KUBERNETES_USER_API_RESPONSE_DEFAULT_USER_ATTR_NOT_FOUND=CWWKS5381W: The social
...
+JSON_MISSING_KEY.useraction=Check the JSON data to determine whether the key is missing.
+
+# 0=JSON key string, 1=JSON value type, for example, object, string, number, and so on, 2=JSON value type, 3=JSON data
+JSON_ENTRY_WRONG_JSON_TYPE=CWWKS5386E: The value for the key [{0}] in the JSON data is expected to be of type {1}, but the value is of type {2}. The JSON data is [{3}].
```
Comment (helyarp) [inline dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops]: Please change "The value for the key [{0}] in the JSON data is expected to be of type {1}" to "The value for the [{0}] key in the JSON data must be of type {1}"

**Comment by ayoho** [comment]
#run-libby-bot

**Pull request merged**